hpx/runtime/components/server/runtime_support.hpp

```diff
@@ -832,6 +835,11 @@ namespace hpx { namespace components { namespace server
         agas::unmark_as_migrated(id);
 
         to_migrate.make_unmanaged();
+
+        // inform the newly created component that it has been migrated
+        if (new_instance != nullptr)
+            new_instance->on_migrated();
```

Comment:
I think this should be called before `agas::unmark_as_migrated`. There could be another migration process invalidating `new_instance`